gpflow/optimizers/scipy.py

```diff
@@ -134,8 +138,9 @@ def assign_tensors(to_tensors: Iterable[tf.Variable], values: Iterable[tf.Tensor
 V = TypeVar("V", bound=Variables)
 
 
-def _compute_loss_and_gradients(loss_cb: LossClosure, variables: V) -> Tuple[tf.Tensor, V]:
-    with tf.GradientTape() as tape:
-        loss = loss_cb()
-    grads = tape.gradient(loss, variables)
+def _compute_loss_and_gradients(loss_closure: LossClosure, variables: V) -> Tuple[tf.Tensor, V]:
+    with tf.GradientTape(watch_accessed_variables=False) as tape:
+        tape.watch(variables)
+        loss = loss_closure()
+        grads = tape.gradient(loss, variables)
```

Comment:
```suggestion
    grads = tape.gradient(loss, variables)
```

This is not a speed-up, but slow-down ;)